sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/models/one/DeviceExtension.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.appcenter.ingestion.models.one;
+
+import com.microsoft.appcenter.ingestion.models.Model;
+import com.microsoft.appcenter.ingestion.models.json.JSONUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONStringer;
+
+/**
+ * The “device” extension tracks common device elements that are not available in the core envelope.
+ */
+public class DeviceExtension implements Model {
+
+    /**
+     * Local ID property.
+     */
+    private static final String LOCAL_ID = "localId";
+
+    /**
+     * Local ID.
+     */
+    private String localId;
+
+    /**
+     * Get local ID.
+     *
+     * @return local ID.
+     */
+    public String getLocalId() {
+        return localId;
+    }
+
+    /**
+     * Set local ID.
+     *
+     * @param localId local ID.
+     */
+    public void setLocalId(String localId) {
+        this.localId = localId;
+    }
+
+    @Override
+    public void read(JSONObject object) {
+        setLocalId(object.optString(LOCAL_ID, null));
+    }
+
+    @Override
+    public void write(JSONStringer writer) throws JSONException {
+        JSONUtils.write(writer, LOCAL_ID, getLocalId());
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Do you have code analysis warning simplifiable if statement?

Reply:
Not for this one. It has nothing to simplify.